### ISSUE TITLE
Add Header for User ID

### ISSUE THREAD

**Issue by lbreitk** [greenpau/caddy-auth-jwt#58]
Right now, there are headers for User Name, User Email, User Roles, and User Subject, but no header for the User ID. Currently, I am authenticating via oauth, which is authed against LDAP, and logging the user into netbox by passing a header. I'd prefer to use the ID they use for LDAP at the ID to identify them in netbox, however that doesn't seem to be an option.

**Comment by greenpau** [comment]
@lbreitk , got it. Please ping back in a couple of weeks. Hopefully, by then I will finish my refactoring and then we can add the things you need.

it may be that the you thought about the solution solved your particular issue, but may have been an issue for others.

Consider thinking in terms of caddyfile directives. For example, here are some directives that I thought about.

```
jwt {
  inject header X-HEADER from claim email
  inject header X-USER from claim metadata.user
}
```

Additionally, please think about testing. If you propose a change, please provide unit tests.

To sum up, when you are ready, please open a PR to implement the above injections and associated unit tests.

Please do not hesitate asking question! Any question is a gem :-)

**Comment by greenpau** [comment]
@lbreitk , I am ready to engage on this one. 

Think in terms of JWT token fields and provide the instructions you want to see.

```
jwt {
  inject header X-HEADER from claim email
  inject header X-USER from claim metadata.user
}
```

Currently, here is what is being passed: https://github.com/greenpau/caddy-auth-jwt#pass-token-claims-in-http-headers

**Comment by lbreitk** [comment]
@greenpau Thanks for the reply! Unfortunately, I'm a Systems Engineer - my developers skills may be a bit limited to be of much more help :P But I will try to think of a good comprehensive reply :)

**Comment by phss** [comment]
Hey @greenpau and @lbreitk I can give it a try, if that's okay, as this is exactly what I need. I am not super experienced in Go, but sounds fun anyway. :D

**Comment by greenpau** [comment]
FYI, starting working on this issue.

**Comment by greenpau** [comment]
@lbreitk , @phss , this is now working. See [docs](https://github.com/greenpau/caddy-auth-jwt#custom-headers).
